api/src/main/scala/dx/api/DxIoParameter.scala

```diff
@@ -75,21 +75,22 @@ final case class IOParameterValueDataObject(id: String,
     dxApi.dataObject(id, project.map(dxApi.project))
   }
 }
+final case class DxIoParameterValueReference(fields: Map[String, String]) extends IOParameterValue
 final case class DxIoParameterValuePath(project: String,
                                         path: String,
                                         name: Option[String],
                                         region: Option[String] = None)
     extends IOParameterValue
 
 object IOParameterValue {
-  def forFileField(field: String,
-                   id: Option[String],
-                   project: Option[String] = None,
-                   name: Option[String] = None,
-                   path: Option[String] = None,
-                   region: Option[String] = None): IOParameterValue = {
+  def forDataObjectField(field: String,
+                         id: Option[String],
+                         project: Option[String] = None,
+                         name: Option[String] = None,
+                         path: Option[String] = None,
+                         region: Option[String] = None): IOParameterValue = {
     field match {
-      case DxIOSpec.Default if id.isDefined && Seq(name, project, path, region).forall(_.isEmpty) =>
+      case DxIOSpec.Default if id.isDefined && Seq(name, path, region).forall(_.isEmpty) =>
         IOParameterValueDataObject(id.get)
       case DxIOSpec.Default =>
         throw new Exception("default file value may not have name, project, or path")
```

Comment:
Re-word this error message if project is allowed?

Reply:
good catch